apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -153,17 +196,53 @@ def native_context(self) -> clib.SOMAContext:
         """The C++ SOMAContext for this SOMA context."""
         with self._lock:
             if self._native_context is None:
-                if self._initial_config is None:
-                    self._native_context = _default_global_native_context()
-                else:
+                # SOMAContext needs to be lazily created
+                if self._initial_config is not None:
+                    # The user passed in a tiledb_config
                     cfg = self._internal_tiledb_config()
                     self._native_context = clib.SOMAContext(
                         {k: str(v) for k, v in cfg.items()}
                     )
-            return self._native_context
+                elif self._tiledb_ctx is not None:
+                    # The user passed in a tiledb_ctx; if tiledb is not installed
+                    # it should be impossible to enter into this block because
+                    # we already check that in the constructor
+                    assert tiledb is not None
+                    cfg = self._tiledb_ctx().config().dict()
```

Comment:
I don't think `_tiledb_ctx` is a method, so I suspect this should drop the parens, e.g.,
> cfg = self._tiledb_ctx.config().dict()

Reply:
ps. assuming my comment is correct, does this imply a missing test case?  (or my comment is wrong!)

Reply:
You are right that there should not be parens.

The missing test case brings up a another issue though. We have removed all tiledb-py usage from the unit tests. But should we temporarily re-introduce it for testing with `tiledb.Ctx()`? We would `xskip` the test if `tiledb` is not installed locally. But then do we need to install `tiledb` for CI runs? (cc @johnkerl)

Reply:
yeah, tough one.  I don't see any great solution - options?

The best I can come up with is:
1. Code the tests to work both with and without the package installed:
    * if installed, test that it works as expected
    * if not installed, test that it generates the right error
2. Have two CI's:  one with it installed, one with it not installed

That at least mimics real-world conditions.